### PR TITLE
add docker image

### PR DESCRIPTION
**Pull request by bclermont**
docker is used to build and test, not to run. here is an image you can publish

note that it use multi stage docker, it need at least docker 17.05 to build (but not execute)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -193,6 +193,23 @@ $ $EDITOR ~/.dcrd/dcrd.conf
...
+- `/etc/dcrwallet` **mandatory** hold `config`, `rpc.cert ` and `rpc.key`
+- `/var/log/dcrwallet` is where the log files are stored
+- `/var/lib/dcrwallet` **important** hold the private keys, don't keep that volume ogn a temporary file system.
+
+Ports:
+
+- `9110` gRPC
+- `9111` Legacy RPC
+
+Any command passed to the container are `btcwallet` arguments, like `--help`.
```
Comment (jrick) [inline README.md]: dcrwallet

File: README.md [hunk 2]
```diff
@@ -193,6 +193,23 @@ $ $EDITOR ~/.dcrd/dcrd.conf
...
+Volumes:
+
+- `/etc/dcrwallet` **mandatory** hold `config`, `rpc.cert ` and `rpc.key`
+- `/var/log/dcrwallet` is where the log files are stored
+- `/var/lib/dcrwallet` **important** hold the private keys, don't keep that volume ogn a temporary file system.
+
+Ports:
+
+- `9110` gRPC
+- `9111` Legacy RPC
```
Comment (jrick) [inline README.md]: these are reversed, 9110/19110 is the json-rpc (legacy) port

**Comment by jrick** [comment]
@jholdstock can you take a look at this? docker isn't my expertise.

File: README.md (hunk 1 above)
Comment (bclermont) [reply README.md]: Fixed https://github.com/decred/dcrwallet/pull/999/commits/de81ed27e29527d6e736294674db967ae32f35a6

Sorry, I did some work on btcwallet before you forked it and the name stick in my brain :)

File: README.md (hunk 2 above)
Comment (bclermont) [reply README.md]: fixed https://github.com/decred/dcrwallet/pull/999/commits/a0bf967daf7cdff5fa492f8e04b5e5f400180aed

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,19 @@
+FROM decred/decred-golang-builder-1.9 as builder
```
Comment (jholdstock) [inline Dockerfile]: I believe `dcrwallet` should be built with go 1.10 or 1.11.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+COPY . /go/src/github.com/decred/dcrwallet
+WORKDIR /go/src/github.com/decred/dcrwallet
+RUN dep ensure
```
Comment (jholdstock) [inline Dockerfile]: I'm not a golang dev so I could be wrong, but following the #dev channel in Slack leads me to believe that the project doesn't use `dep` any more

**Comment by bclermont** [comment]
I gave up a long time ago to contribute here

**Pull request closed without merging**